### ISSUE TITLE
allow to pass single Char to starts(Not)With/ends(Not)With

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#66]
*Platform* (JVM, JS, android): all

## Code related feature
check if a CharSequence startsWith/endsWith a certain Char should also be supported next to CharSequence.
```
assert("hello world").startsWith('h')
```

Add overloads to startsWith, startsNotWith, endsWith and endsNotWith in:
- api-fluent-en_GB-common

which expects a `Char` and delegates to the existing function by converting it to a `String`.
(the functions are located in files named charSequenceAssertions.kt)

# Your first contribution?
- write a comment if you would like to take this task over, I might give you some hints in addition or update the task in case things have changed in the meantime
- do not hesitate to contact me via slack if you need help: https://kotlinlang.slack.com/team/U3DE1TXKP ([Invite yourself](http://slack.kotlinlang.org/) in case you do not have an account yet)

**Comment by kssc0112** [comment]
If overload startsWith like 
``` fun startsWith(subjectProvider: SubjectProvider<CharSequence>, expected: Char): Assertion```
I get the following error in the spec. Is there a workaround for this? If not, I'll continue with creating startsWithChar, startsNotWithChar and so on.

<img width="1239" alt="Screen Shot 2019-07-17 at 11 03 02 PM" src="https://user-images.githubusercontent.com/15913002/61433098-3d3f9680-a8e7-11e9-80e8-f714b9a933b1.png">

**Comment by robstoll** [comment]
there is a workaround, see for instance ListFeatureAssertionsSpec

**Comment by robstoll** [comment]
roughly:
```
import kotlin.reflect.KFunction1
class CharSequenceAssertionsSpec : ch.tutteli.atrium.specs.integration.CharSequenceAssertionsSpec(
    ...
    startsWithChar.name to startsWithChar
){
    companion object{
        val startsWithChar: KFunction1<Expect<CharSequence>, Char, Expect<CharSequence>> = Expect<CharSequence>::get
    }
}
```

**Comment by kssc0112** [comment]
@robstoll Thanks. Almost done for a PR. What is the purpose of calling forSubjectLess and forChecking APIs in CharSequenceAssertionsSpec? Do we want it for the Char versions as well?

Ex. `startsWithPair.forChecking("a", "abc", "xyz")`
`startsWithPair.forSubjectLess("")`

**Comment by robstoll** [comment]
those are two basic specs which every assertion function should hold -- so yes, please add it also for Char. 

SubjectLessSpec tests that the assertion function does not throw an exception even if the subject is not defined. This happens when one transforms the assertion where the transformation as such can fail. For instance:
```
expect(listOf(1,2)).get(100) { 
  // subject of the assertion has now type Int but is effectively not available because index 100 is out of bound
  isGreaterThan(1)
  isLessThan(2)
}
```

CheckingAssertionSpec checks that the assertion does not throw an AssertionError on its own

**Comment by kssc0112** [comment]
Hi Robert,

My PR is ready for your initial review: https://github.com/robstoll/atrium/pull/105

For some reason, I read the last part of your comment only after completing the code.
> which expects a Char and delegates to the existing function by converting it to a String.
(the functions are located in files named charSequenceAssertions.kt)

Please let me know if my changes misalign with your requirement. If so, I'll work on a revision.

**Comment by robstoll** [comment]
implemented with 850bd77
